catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/DeleteOperations.java

```diff
@@ -260,6 +267,54 @@ private DeleteResponse performLocalDelete(DeleteRequest deleteRequest,
         return deleteResponse;
     }
 
+    //
+    // Private helper methods
+    //
+    private DeleteRequest rewriteQuery(DeleteRequest deleteRequest)
+            throws StopProcessingException, UnsupportedQueryException {
+
+        final String attributeName = deleteRequest.getAttributeName();
+        if (Metacard.ID.equals(attributeName)) {
+            return deleteRequest;
+        }
+
+        Filter filter = frameworkProperties.getFilterBuilder()
+                .anyOf(deleteRequest.getAttributeValues()
+                        .stream()
+                        .map(e -> frameworkProperties.getFilterBuilder()
+                                .attribute(attributeName)
+                                .is()
+                                .equalTo()
+                                .text(e.toString()))
+                        .collect(Collectors.toList()));
+        SourceResponse response = sourceOperations.getCatalog().query(new QueryRequestImpl(new QueryImpl(filter), false));
```

Comment:
Did you test a scenario where the response would be larger than the default page size of 20?  I am afraid if you have more than 20 history and non-history results, you might throw the `StopProcessingException`.